### ISSUE TITLE
[Feature Request] Support for file:// protocol to use it as json URL pointing to local filesystem file

### ISSUE THREAD

**Issue by kuolemaaa** [dario-baumberger/obsidian-json-table#6]
**Is your feature request related to a problem? Please describe.**
file:// is not supported. it could be very handy.
console reports:
```
(node:347398) UnhandledPromiseRejectionWarning: Error: ClientRequest only supports http: and https: protocols
    at new ClientRequest (node:electron/js2c/browser_init:2:111711)
    at Object.request (node:electron/js2c/browser_init:2:48365)
    at IpcMainImpl.<anonymous> (/usr/lib/obsidian/obsidian.asar/main.js:38:7188)
    at IpcMainImpl.emit (node:events:519:28)
    at WebContents.<anonymous> (node:electron/js2c/browser_init:2:82518)
    at WebContents.emit (node:events:519:28)
(Use `electron --trace-warnings ...` to show where the warning was created)
(node:347398) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)

```

**Describe the solution you'd like**
¯\\\_(ツ)_/¯ support file:// for the "Generate table from selected JSON URL", i guess

**Describe alternatives you've considered**
No alternative considered

**Comment by dario-baumberger** [comment]
Many thanks for the feature request. I'll take a look at Obsidian's file adapters and see what's possible. 

I've started very basic with "in editor JSON", but I can see that file system paths could be useful.

**Comment by dario-baumberger** [comment]
Due to limitations of Obsidian (more precisely, Electron), this is not possible. Or at least, I was not able to do it. :)

**Comment by kuolemaaa** [comment]
quick n dirty in 30ish minutes. i didnt even know ts nor obsidian api. feel free to continue. consider it as a poc.

https://github.com/dario-baumberger/obsidian-json-table/pull/8

**Comment by dario-baumberger** [comment]
@kuolemaaa thanks for your effort!

In your PR, you're solving the problem inside the vault, which is a different case from what’s described here 😄. Inside the vault, Obsidian’s API provides full access, so that works fine.

The original request was:  _Support for file:// protocol to use it as json URL pointing to local filesystem file_. Since Obsidian runs on Electron, there are restrictions on accessing the local file system outside the vault.

https://www.electronjs.org/docs/latest/tutorial/security#18-avoid-usage-of-the-file-protocol-and-prefer-usage-of-custom-protocols

**Comment by kuolemaaa** [comment]
Yeah, I know. but better than nothing. still lacking (imho) critical feature of feeding it from file, even with restriction
